apps/drupal/src/Plugin/Preprocessor/README.md

```diff
@@ -0,0 +1,51 @@
+# Particle Preprocessor
+
+The Particle Preprocessor Plugin creates a system to preprocess theme templates.
+The system allows for processing via PHP rather than within Twig templates or
+overly long `.theme` files.
+
+## Preprocessors
+
+The Preprocessor directory contains the PHP work to preprocess templates.
+To match the Plugin system in Drupal core, these files should be named in the following manner:
+
+Entity - Bundle - View Mode 
+
+The Preprocessor directory is automatically scanned and loaded as a Class Map in ParticlePreprocessorManager.php.
+
+For now, each entity, bundle and view name should be separated by a capital letter.
+For example, if you have a `node--article--card-max`, name your php file: `NodeArticleCardmax.php`
+
+The directory scanner creates a suggestions => preprocessors matching array:
+
+```
+entity__bundle__view => EntityBundleView
+```
+
+There are three Preprocessors provided to the theme by default:
+
+* `Base.php`
+* `Page.php`
+* `Node.php`
+
+The `Base.php` preprocessor is the root implementation all preprocessors inherit from.
+It represents the base implementation of a preprocessor.
+
+`Page.php` and `Node.php` preprocess their respective hooks via the PreprocessManager method:
+
+See `particle_preoprocess_page` hook to see how this is added:
+
+```php
+/**
+ * Implements template_preprocess_page().
+ */
+function particle_preprocess_page(&$variables) {
+  // Process All Variables Via PreprocessClass.
+  PreprocessorManager::loadByVariables('page', $variables)
+    ->process($variables);
+}
+```
+
+With the above in place as the hook, `loadByVariables` will direct hook type
```

Comment:
That makes sense for this simple example, but what about nodes of specific type and display modes? Do I need to write the preprocess myself to pull those files in, or are you saying it happens automatically just by naming convention? If that's the case, why is it needed manually for Pages?

Reply:
Each type is only "turned on" by placing it on the hook you want to use it for. Because it's essentially mapping the hooks and their theme suggestions to the classes you create and rerouting them to that map. `loadByVariables` here says, "okay give me all your page suggestions, and load them into a class if it's available (e.g., a file with that name exists)."

Reply:
Maybe it would make sense for particle.theme to have the preprocess_page and preprocess_node implementations in place, but commented out, so that it's a bit more obvious how to start using it?

Reply:
I added how to turn it on for Node and Page to the README. Figured that'd be easiest!